packages/x-date-pickers/src/DateCalendar/tests/DateCalendar.spec.tsx

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+import { DateCalendar } from '@mui/x-date-pickers/DateCalendar';
+
+<DateCalendar />;
+
+<DateCalendar defaultValue={new Date()} />;
```

Comment:
The change in this PR are only for standalone usage.
When `DateCalendar` (or the other) as used inside a picker, they always receive a value even on the new pickers because there is a `useControlled` higher in the chain.

Reply:
Oh I see what you mean, the `defaultValue` prop should not leak on the old pickers

Reply:
If it works—then it's probably not a problem, it's just that as far as I've tested a few instances—it did not seem to work correctly. 🤔

Reply:
It does not work and `value` is required anyway on the old pickers so the fallback would never occur :+1: